.devcontainer/Mojo.Dockerfile

```diff
@@ -0,0 +1,179 @@
+# ===----------------------------------------------------------------------=== #
+# Copyright (c) 2024, Olivier Benz
+# Copyright (c) 2024, Modular Inc. All rights reserved.
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions:
+# https://llvm.org/LICENSE.txt
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# The Dockerfile is provided as reference. Please review the redistribution
+# terms of the Mojo license in Section 1 (https://www.modular.com/legal/mojo)
+# prior to distributing pre-built container images.
+# ===----------------------------------------------------------------------=== #
+
+ARG BUILD_ON_IMAGE=glcr.b-data.ch/mojo/base
+ARG MOJO_VERSION=nightly
+ARG LLVM_VERSION
+
+FROM ${BUILD_ON_IMAGE}:${MOJO_VERSION} as mojo
+
+ARG DEBIAN_FRONTEND=noninteractive
+
+ARG BUILD_ON_IMAGE
+ARG LLVM_VERSION
+
+ENV PARENT_IMAGE=${BUILD_ON_IMAGE}:${MOJO_VERSION} \
+    PARENT_IMAGE_BUILD_DATE=${BUILD_DATE} \
+    LLVM_VERSION=${LLVM_VERSION}
+
+ENV PATH=${LLVM_VERSION:+/usr/lib/llvm-}${LLVM_VERSION}${LLVM_VERSION:+/bin:}$PATH
+
+RUN dpkgArch="$(dpkg --print-architecture)" \
+  ## Ensure that common CA certificates
+  ## and OpenSSL libraries are up to date
+  && apt-get update \
+  && apt-get -y install --no-install-recommends --only-upgrade \
+    ca-certificates \
+    openssl \
+## Install Modular
+  && apt-get -y install --no-install-recommends \
+    apt-transport-https \
+  && curl -s https://dl.modular.com/public/installer/setup.deb.sh | bash - \
+  && apt-get -y install --no-install-recommends \
+    modular \
+  && if [ "$MOJO_VERSION" = "nightly" ]; then \
+    ## Update Mojo nightly
+    modular update nightly/mojo; \
+  fi \
+  ## Clean up
+  && rm -rf /root/.cache \
+    /root/.ipython \
+    /root/.lldb \
+    /root/.local \
+## Install tools for buildig and testing the Mojo standard library
+  && if [ -n "$LLVM_VERSION" ]; then \
+    ## Install LLVM
+    apt-get -y install --no-install-recommends \
+      lsb-release \
+      software-properties-common; \
+    curl -sSLO https://apt.llvm.org/llvm.sh; \
```

Comment:
**Suggestion** Is it possible to use the same script CI is using (https://github.com/modularml/mojo/blob/405e369b84ce32fe87f617b210eef81867371bdf/stdlib/scripts/install-build-tools-linux.sh) for downloading LLVM and setting up the various alternatives?  It may lead to some reduced maintenance/drift over time.  I think we'll just want to teach https://github.com/modularml/mojo/blob/405e369b84ce32fe87f617b210eef81867371bdf/stdlib/scripts/install-build-tools-linux.sh to take an argument for the LLVM version and specify it here in the Dockerfile.